[CI] IndexingIT » testIndexing (and others) failing

Multiple tests in :x-pack:qa:rolling-upgrade:v7.12.0#oneThirdUpgradedTest time out while waiting for a healthy cluster:

:x-pack:qa:rolling-upgrade:v7.12.0#oneThirdUpgradedTest | FAILED | 3m 48.581s |  
  | org.elasticsearch.upgrades.DataStreamsUpgradeIT | FAILED | 1m 10.341s |  
  |  | testDataStreamValidationDoesNotBreakUpgrade | FAILED | 1m 9.823s |  
  | org.elasticsearch.upgrades.IndexingIT | FAILED | 1m 10.097s |  
  |  | testIndexing | FAILED | 1m 10.097s |  
  | org.elasticsearch.upgrades.SearchableSnapshotsRollingUpgradeIT | FAILED | 31.386s |  
  |  | testBlobStoreCacheWithPartialCopyInMixedVersions | FAILED | 30.397s |  
  | org.elasticsearch.upgrades.RollupDateHistoUpgradeIT | FAILED | 29.810s |  
  |  | testDateHistoIntervalUpgrade | FAILED | 29.810s

<br class="Apple-interchange-newline">

**Build scan**:
https://gradle-enterprise.elastic.co/s/6qtdxensr4ulk

**Repro line**:
`./gradlew ':x-pack:qa:rolling-upgrade:v7.12.1#oneThirdUpgradedTest' -Dtests.class="org.elasticsearch.upgrades.SearchableSnapshotsRollingUpgradeIT" -Dtests.method="testBlobStoreCacheWithPartialCopyInMixedVersions" -Dtests.seed=AFE02501F5B723CA -Dtests.bwc=true -Dtests.locale=sr -Dtests.timezone=GMT0 -Druntime.java=8`

**Reproduces locally?**:
yes

**Applicable branches**:
7.16

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P28D&search.timeZoneId=Europe/Zurich&tests.container=org.elasticsearch.upgrades.IndexingIT&tests.sortField=FAILED&tests.test=testIndexing&tests.unstableOnly=true

**Failure excerpt**:
```
java.lang.AssertionError: timed out waiting for green state for index [] cluster state [{ |  
-- | --
  | "cluster_name" : "v7.12.0", |  
  | "cluster_uuid" : "ApCjwrAARKWpo8d2NEkFJg", |  
  | "version" : 739, |  
  | "state_uuid" : "TvzfBxtTSimW0Hlro0JsGQ", |  
  | "master_node" : "TyArAitPQgiwcPCcXnwRXQ", |  
  | "blocks" : {

...

at __randomizedtesting.SeedInfo.seed([4B479E4042C139B:E06E36ED22595010]:0) |  
-- | --
  |   | at org.junit.Assert.fail(Assert.java:88) |  
  |   | at org.elasticsearch.test.rest.ESRestTestCase.ensureHealth(ESRestTestCase.java:1310) |  
  |   | at org.elasticsearch.test.rest.ESRestTestCase.ensureHealth(ESRestTestCase.java:1298) |  
  |   | at org.elasticsearch.test.rest.ESRestTestCase.ensureHealth(ESRestTestCase.java:1294) |  
  |   | at org.elasticsearch.upgrades.IndexingIT.testIndexing(IndexingIT.java:40) |  
  |   | at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) |  
  |   | at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) |  
  |   | at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) |  
  |   | at java.lang.reflect.Method.invoke(Method.java:498) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49) |  
  |   | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  |   | at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48) |  
  |   | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  |   | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368) |  
  |   | at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817) |  
  |   | at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894) |  
  |   | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41) |  
  |   | at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) |  
  |   | at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53) |  
  |   | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) |  
  |   | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  |   | at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368) |  
  |   | at java.lang.Thread.run(Thread.java:748)
```

--- a/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SearchableSnapshotsRollingUpgradeIT.java
+++ b/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SearchableSnapshotsRollingUpgradeIT.java
@@ -16,6 +16,7 @@ import org.elasticsearch.client.Request;
 import org.elasticsearch.client.Response;
 import org.elasticsearch.client.searchable_snapshots.MountSnapshotRequest.Storage;
 import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.routing.allocation.DataTier;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.xcontent.support.XContentMapValues;
@@ -114,7 +115,6 @@ public class SearchableSnapshotsRollingUpgradeIT extends AbstractUpgradeTestCase
         executeBlobCacheCreationTestCase(storage, 9876L);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/79786")
     public void testBlobStoreCacheWithPartialCopyInMixedVersions() throws Exception {
         final Storage storage = Storage.SHARED_CACHE;
         assumeVersion(Version.V_7_12_0, Storage.SHARED_CACHE);
@@ -392,6 +392,11 @@ public class SearchableSnapshotsRollingUpgradeIT extends AbstractUpgradeTestCase
         final Request request = new Request(HttpPost.METHOD_NAME, "/_snapshot/" + repositoryName + '/' + snapshotName + "/_mount");
         if (UPGRADE_FROM_VERSION.onOrAfter(Version.V_7_12_0)) {
             request.addParameter("storage", storage.storageName());
+            if (UPGRADE_FROM_VERSION.before(Version.V_7_13_0) && storage == Storage.SHARED_CACHE) {
+                // Force tier preference to frozen since 7.12.x since 7.12.x will otherwise try to use the tier preference of frozen
+                // followed by all other tiers by default which 7.13+ nodes will not accept.
+                indexSettings = Settings.builder().put(indexSettings).put(DataTier.TIER_PREFERENCE, DataTier.DATA_FROZEN).build();
+            }
         } else {
             assertThat("Parameter 'storage' was introduced in 7.12.0 with " + Storage.SHARED_CACHE, storage, equalTo(Storage.FULL_COPY));
         }

